confgenerator/fluentbit/modular.go

```diff
@@ -43,14 +43,29 @@ func (c Component) generateSection() string {
 	return fmt.Sprintf("[%s]\n%s\n", c.Kind, strings.Join(lines, "\n"))
 }
 
-type Pipeline struct {
+type ModularConfig struct {
+	Variables  map[string]string
 	Components []Component
-	LogName    string
-	Internal   bool
 }
 
-type ModularConfig struct {
-	Pipelines      []Pipeline
-	InternalOutput Component
-	Output         Component
+func (c ModularConfig) Generate() (string, string, error) {
+	var parts []string
+	for k, v := range c.Variables {
+		parts = append(parts, fmt.Sprintf("@SET %s=%s", k, v))
+	}
+	sort.Strings(parts)
+	// Blank line
+	parts = append(parts, "")
+
+	var parserParts []string
+
+	for _, c := range c.Components {
+		out := c.generateSection()
+		if c.Kind == "PARSER" {
+			parserParts = append(parserParts, out)
+		} else {
+			parts = append(parts, out)
```

Comment:
[Optional] To minimize the diffs, it might help to group these by `c.Kind` and emit them in the order `SERVICE`,`INPUT`,`FILTER`,`OUTPUT`.

Reply:
This makes the diff slightly smaller, and I am grumpy about the complexity, but I did it. We can revert in a subsequent cleanup CL.